src/celutil/utf8.h

```diff
@@ -66,6 +66,24 @@ inline int UTF8EncodedSize(wchar_t ch)
 #endif
 }
 
+constexpr inline int UTF8EncodedSizeChecked(std::uint32_t ch)
```

Comment:
For now I don't want to change the existing `UTF8EncodedSize` function which supports up to 6-byte "UTF-8" sequences and is used at various places around the codebase, but I intend to fix that in a later PR.